### ISSUE TITLE
Error when delete text from multi line Entry

### ISSUE THREAD

**Issue by vladimirschuka** [fyne-io/fyne#2129]
When you delete text from any line with key backspace of  widget "Entry" and cursor reached beginning of the line, cursor does not go on up as expected, instead we have an error:

```
panic: runtime error: slice bounds out of range [-1:]

goroutine 34 [running]:
fyne.io/fyne/v2/widget.(*textProvider).deleteFromTo(0xc0002c6750, 0xffffffffffffffff, 0x0, 0x0, 0xc00040e238, 0x8)
        /home/vshchuka/Develop/GOnotPATH/fyne/widget/text.go:170 +0x23e
fyne.io/fyne/v2/widget.(*Entry).TypedKey(0xc0002c5680, 0xc000b40530)
        /home/vshchuka/Develop/GOnotPATH/fyne/widget/entry.go:540 +0xc18
fyne.io/fyne/v2/internal/driver/glfw.(*window).keyPressed.func7()
        /home/vshchuka/Develop/GOnotPATH/fyne/internal/driver/glfw/window.go:1137 +0x38
fyne.io/fyne/v2/internal/driver/glfw.(*window).runEventQueue(0xc00015e540)
        /home/vshchuka/Develop/GOnotPATH/fyne/internal/driver/glfw/window.go:1247 +0x7a
created by fyne.io/fyne/v2/internal/driver/glfw.(*gLDriver).createWindow.func1
        /home/vshchuka/Develop/GOnotPATH/fyne/internal/driver/glfw/window.go:1271 +0x10d
exit status 2

```
To reproduce:

1) take the example 
https://github.com/fyne-io/fyne/tree/master/cmd/fyne_demo
2) go run main.go
3) go to "Text" tab
4) try to enter text in the widget where lorem Ipsum text is.
5) try to delete full string until cursor is in the beginning of the row
6) press backspace couple of times 
7) bug will occur

OS :
Ubuntu 20.04 
go version 1.16.2

**Comment by fpabl0** [comment]
Could you please use the issue template? It contains the necessary information to help us to know more about the problem you are experiencing. Thanks for your understanding.

**Comment by andydotxyz** [comment]
As @fpabl0 said key information was missing, such as the Fyne version.
However I think that a fix for this was just released in `v2.0.2` so please try upgrading to that version and trying again.

**Comment by vladimirschuka** [comment]
I have tried v2.0.1 yesterday. I will try the new one.

**Comment by vladimirschuka** [comment]
Fixed in v 2.0.2
